dashboard/app/views/teacher_dashboard/show.html.haml

```diff
@@ -13,6 +13,7 @@
   teacher_dashboard_data[:localeCode] = request.locale
   teacher_dashboard_data[:textToSpeechScriptIds] = Script.text_to_speech_script_ids
   teacher_dashboard_data[:preReaderScriptIds] = Script.pre_reader_script_ids
+  teacher_dashboard_data[:lessonExtrasScriptIds] = Script.lesson_extras_script_ids
```

Comment:
Curious whether best practice is to set up data to pass to JS here or in the controller? Considered changing to the controller, but looks like we do it in both places in different places in our code base.

Reply:
This seems like a reasonable way to pass the data to me. Which controller are you referring to?

Reply:
Like we could put all of this [here](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/teacher_dashboard_controller.rb#L5), which is what we do for the [Code Studio homepage](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/home_controller.rb#L98), I think. Doesn't really matter I guess, I just thought views should just consume data vs. defining it, so seemed odd to me.

Reply:
Oh I see! Yeah I think that you're right, it should just be the job of the view to do the displaying. It's a weird case because we're just converting the data into a json object to be passed to JS so it's not the traditional view like is described here https://medium.com/stackavenue/m-v-c-architecture-of-ruby-on-rails-9712719a69ed

Reply:
I don't have a strong opinion about whether the data should be converted to a json object here or in the controller, but it would be nice to stick to a pattern. Would be interesting to hear people's opinions, if you decide to ask in a channel, tag me!

Reply:
Yeah totally, we kind of skip the V in MVC :).